### DISCUSSION TITLE
Can not make MacOS App on MacOS Big Sur

### DISCUSSION THREAD

**Discussion by leogitpro** [pyinstaller/pyinstaller#5786] [Help]
Try many many ways, For windows and Linux all platform make the onefile application successfull, but failed on Mac OS Big Sur.

Tested the python version:
3.8.2 apple system from command-line
3.9.4 from brew installed.
pyinstaller: 4.3 latest version.
All of failed.

## Description of the issue

There is the log:

pyinstaller -F -w -y --clean --icon assets/images/logo.icns --add-data assets:assets PyInstallerDemo.py
77 INFO: PyInstaller: 4.3
77 INFO: Python: 3.8.2
84 INFO: Platform: macOS-11.2.3-x86_64-i386-64bit
85 INFO: wrote /Users/leo/GitHub/python-demos/PyInstaller/PyInstallerDemo.spec
88 INFO: UPX is not available.
88 INFO: Removing temporary files and cleaning cache in /Users/leo/Library/Application Support/pyinstaller
103 INFO: Extending PYTHONPATH with paths
['/Users/leo/GitHub/python-demos/PyInstaller',
 '/Users/leo/GitHub/python-demos/PyInstaller']
109 INFO: checking Analysis
109 INFO: Building Analysis because Analysis-00.toc is non existent
109 INFO: Initializing module dependency graph...
110 INFO: Caching module graph hooks...
114 WARNING: Several hooks defined for module 'win32ctypes.core'. Please take care they do not conflict.
117 INFO: Analyzing base_library.zip ...
2136 INFO: Processing pre-find module path hook distutils from '/Library/Python/3.8/site-packages/PyInstaller/hooks/pre_find_module_path/hook-distutils.py'.
2136 INFO: distutils: retargeting to non-venv dir '/Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8'
3403 INFO: Caching module dependency graph...
3534 INFO: running Analysis Analysis-00.toc
3547 INFO: Analyzing /Users/leo/GitHub/python-demos/PyInstaller/PyInstallerDemo.py
3565 INFO: Processing module hooks...
3565 INFO: Loading module hook 'hook-PyQt5.py' from '/Library/Python/3.8/site-packages/PyInstaller/hooks'...
3672 WARNING: Hidden import "sip" not found!
3672 INFO: Loading module hook 'hook-encodings.py' from '/Library/Python/3.8/site-packages/PyInstaller/hooks'...
3721 INFO: Loading module hook 'hook-PyQt5.QtGui.py' from '/Library/Python/3.8/site-packages/PyInstaller/hooks'...
3750 INFO: Loading module hook 'hook-pickle.py' from '/Library/Python/3.8/site-packages/PyInstaller/hooks'...
3752 INFO: Loading module hook 'hook-heapq.py' from '/Library/Python/3.8/site-packages/PyInstaller/hooks'...
3752 INFO: Loading module hook 'hook-difflib.py' from '/Library/Python/3.8/site-packages/PyInstaller/hooks'...
3753 INFO: Loading module hook 'hook-PyQt5.QtWidgets.py' from '/Library/Python/3.8/site-packages/PyInstaller/hooks'...
3790 INFO: Loading module hook 'hook-multiprocessing.util.py' from '/Library/Python/3.8/site-packages/PyInstaller/hooks'...
3791 INFO: Loading module hook 'hook-sysconfig.py' from '/Library/Python/3.8/site-packages/PyInstaller/hooks'...
3799 INFO: Loading module hook 'hook-xml.py' from '/Library/Python/3.8/site-packages/PyInstaller/hooks'...
3990 INFO: Loading module hook 'hook-distutils.py' from '/Library/Python/3.8/site-packages/PyInstaller/hooks'...
3991 INFO: Loading module hook 'hook-PyQt5.QtCore.py' from '/Library/Python/3.8/site-packages/PyInstaller/hooks'...
4014 INFO: Looking for ctypes DLLs
4017 INFO: Analyzing run-time hooks ...
4019 INFO: Including run-time hook '/Library/Python/3.8/site-packages/PyInstaller/hooks/rthooks/pyi_rth_multiprocessing.py'
4021 INFO: Including run-time hook '/Library/Python/3.8/site-packages/PyInstaller/hooks/rthooks/pyi_rth_pyqt5.py'
4026 INFO: Looking for dynamic libraries
4256 INFO: Looking for eggs
4256 INFO: Using Python library /Library/Developer/CommandLineTools/Library/Frameworks/Python3.framework/Versions/3.8/bin/../Python3
4258 INFO: Warnings written to /Users/leo/GitHub/python-demos/PyInstaller/build/PyInstallerDemo/warn-PyInstallerDemo.txt
4285 INFO: Graph cross-reference written to /Users/leo/GitHub/python-demos/PyInstaller/build/PyInstallerDemo/xref-PyInstallerDemo.html
4295 INFO: Appending 'datas' from .spec
4305 INFO: checking PYZ
4305 INFO: Building PYZ because PYZ-00.toc is non existent
4305 INFO: Building PYZ (ZlibArchive) /Users/leo/GitHub/python-demos/PyInstaller/build/PyInstallerDemo/PYZ-00.pyz
4676 INFO: Building PYZ (ZlibArchive) /Users/leo/GitHub/python-demos/PyInstaller/build/PyInstallerDemo/PYZ-00.pyz completed successfully.
4682 INFO: checking PKG
4683 INFO: Building PKG because PKG-00.toc is non existent
4683 INFO: Building PKG (CArchive) PKG-00.pkg
6134 ERROR: Can not find path @executable_path/../Python3 (needed by /Library/Developer/CommandLineTools/usr/bin/python3)
15654 INFO: Building PKG (CArchive) PKG-00.pkg completed successfully.
15658 INFO: Bootloader /Library/Python/3.8/site-packages/PyInstaller/bootloader/Darwin-64bit/runw
15658 INFO: checking EXE
15659 INFO: Building EXE because EXE-00.toc is non existent
15659 INFO: Building EXE from EXE-00.toc
15659 INFO: Appending archive to EXE /Users/leo/GitHub/python-demos/PyInstaller/dist/PyInstallerDemo
15674 INFO: Fixing EXE for code signing /Users/leo/GitHub/python-demos/PyInstaller/dist/PyInstallerDemo
15676 INFO: Building EXE from EXE-00.toc completed successfully.
15680 INFO: checking BUNDLE
15680 INFO: Building BUNDLE because BUNDLE-00.toc is non existent
15680 INFO: Building BUNDLE BUNDLE-00.toc
15698 INFO: moving BUNDLE data files to Resource directory

**Comment by rokm** [comment]
> All of failed.

In what way? According to build log you posted, the build succeeded.

**Reply by leogitpro** [reply]
6134 ERROR: Can not find path @executable_path/../Python3 (needed by /Library/Developer/CommandLineTools/usr/bin/python3)

The nis not build successed. 

A successed build the last will print build successful information.
These stop the moving BUNDLE data files to Resource directory line.

The dist directory app can not open it.

**Comment by rokm** [comment]
>6134 ERROR: Can not find path @executable_path/../Python3 (needed by /Library/Developer/CommandLineTools/usr/bin/python3)

That error is non-fatal, though, as the build process goes on.

> 15676 INFO: Building EXE from EXE-00.toc completed successfully.

The executable was built successfully...

> 15680 INFO: checking BUNDLE
15680 INFO: Building BUNDLE because BUNDLE-00.toc is non existent
15680 INFO: Building BUNDLE BUNDLE-00.toc
15698 INFO: moving BUNDLE data files to Resource directory

But because you're building in non-windowed mode, you're also getting an .app bundle, whose last build message is always `moving BUNDLE data files to Resource directory`.

**Reply by rokm** [reply]
> The dist directory app can not open it.

That's not a build issue, then, but a run-time one.

You should first try running the non-app-bundle executable from terminal to see if that works (because if it doesn't, you'll at least get an error message printed in the terminal). If you're still having problems with .app bundle after that, try adding top-level exception handling in your program, and log exception to file to see what's wrong.
